bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/CachedCommand.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+
+/**
+ * Basic command interface allowing to execute the command.
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public class CachedCommand<ResponseType extends Response<?>> implements Command<ResponseType> {
+
+  private Command<ResponseType> cachedCommand;
+  private ExpiringCache<ResponseType> cache;
+
+  public CachedCommand(Command<ResponseType> cachedCommand) {
+    this(cachedCommand, 1000);
+  }
+
+  public CachedCommand(Command<ResponseType> cachedCommand, int expiry) {
+    this.cachedCommand = cachedCommand;
+    this.cache = new ExpiringCache<ResponseType>(expiry, () -> {
+      try {
+        return this.cachedCommand.execute();
+      } catch (ResponseException | IOException | AuthenticationException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Override
+  public ResponseType execute() throws ResponseException, IOException, AuthenticationException {
+    ExpiringCache<ResponseType> cache = this.cache;
+    try {
+      @Nullable
```

Comment:
Put the @Nullable on the same line as the variable delcaration

Reply:
I'll only have access to my dev environment at the end of next week. I've had a quick look at the @nullables on local variables before and I found that only this one right here is actually required.

Currently I do not have access to my dev environment, I'll remove the other ones sometime around end of next week.

Reply:
From my understanding locals are always nullable.

Reply:
I thought the same, but this @Nullable makes a difference. Without it, it does not even compile:
```
[ERROR] [...]/openhab2-addons/bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/CachedCommand.java:[52,1679] Null type mismatch (type annotations): required 'ResponseType extends @NonNull Response<?>' but this expression has type '@Nullable ResponseType extends @NonNull Response<?>'
```

Unless somebody can tell me a more elegant way to solve this problem, I suppose we should keep it as is (also on a separate line, as this is how the ESH formatter wants it)